### ISSUE TITLE
Unexpected error running Liquibase: ERROR: syntax error at or near "IMMUTABLE"

### ISSUE THREAD

**Issue by venugopal60** [liquibase/liquibase#5258]
### Search first

- [X] I searched and no similar issues were found

### Description

```
<?xml version="1.1" encoding="UTF-8" standalone="no"?>
<databaseChangeLog logicalFilePath="ab.xml" xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
    <changeSet author="test" id="ab">
        <sql splitStatements="false">
CREATE OR REPLACE FUNCTION my_to_date(dt text)
          RETURNS date IMMUTABLE AS $$
          BEGIN
              IF LENGTH(dt) > 24 THEN
                  RETURN to_date(dt, 'Dy Mon DD YYYY') IMMUTABLE;
              ELSE
                  RETURN to_date(dt, 'YYYY-MM-DD') IMMUTABLE;
              END IF;
          END;
          $$ LANGUAGE plpgsql;
          CREATE INDEX IF NOT EXISTS to_date_dt ON registrations
USING BTREE (reg_to_date(dt));

        </sql>
    </changeSet>
</databaseChangeLog>
```
here is the error
```
Unexpected error running Liquibase: ERROR: syntax error at or near "IMMUTABLE"
  Position: 217 [Failed SQL: (0) CREATE OR REPLACE FUNCTION my_to_date(dt text)
          RETURNS date IMMUTABLE AS $$
          BEGIN
              IF LENGTH(dt) > 24 THEN
                  RETURN to_date(dt, 'Dy Mon DD YYYY') IMMUTABLE;
              ELSE
                  RETURN to_date(dt, 'YYYY-MM-DD') IMMUTABLE;
              END IF;
          END;
          $$ LANGUAGE plpgsql;]
```

### Steps To Reproduce

```
<?xml version="1.1" encoding="UTF-8" standalone="no"?>
<databaseChangeLog logicalFilePath="ab.xml" xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
    <changeSet author="test" id="ab">
        <sql splitStatements="false">
CREATE TABLE IF NOT EXISTS mytab
(
    id text COLLATE pg_catalog."default" NOT NULL,
    dt text COLLATE pg_catalog."default",
    CONSTRAINT registrations_pkey PRIMARY KEY (id)
);
CREATE OR REPLACE FUNCTION my_to_date(dt text)
          RETURNS date IMMUTABLE AS $$
          BEGIN
              IF LENGTH(dt) > 24 THEN
                  RETURN to_date(dt, 'Dy Mon DD YYYY') IMMUTABLE;
              ELSE
                  RETURN to_date(dt, 'YYYY-MM-DD') IMMUTABLE;
              END IF;
          END;
          $$ LANGUAGE plpgsql;
          CREATE INDEX IF NOT EXISTS to_date_dt ON mytab
USING BTREE (my_to_date(dt));

        </sql>
    </changeSet>
</databaseChangeLog>
```

### Expected/Desired Behavior

it should create index with immutable functions.

### Liquibase Version

4.4.0

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)

**Comment by kevin-atx** [comment]
@venugopal60 - what database platform are you using?

**Comment by tati-qalified** [comment]
Hi @venugopal60, I wasn't able to reproduce this issue. Here's what I did:

1. Ran only the SQL code that was provided in the changelog - no issues
2. With the provided changelog, ran `updateSql` and ran the generated SQL script - no issues
3. Directly ran `liquibase update` - no issues

There may be something else blocking you, maybe some end delimiter configuration, but it doesn't appear to be a Liquibase issue.

I'll be closing this ticket, but feel free to reach out if you have any other questions.
Thanks,
Tatiana
